### ISSUE TITLE
Endpoint freezes after ExecutionException: Boxed Error

### ISSUE THREAD

**Issue by alexandru** [http4s/http4s#2259]
Not sure if this error was reported yet, but the endpoint freezes and is unable to process more requests:

https://gist.github.com/alexandru/76a1bfed8fda656421248f7ad7a8fc60

Note that only the endpoints that are actually doing something are freezing, because the homepage that simply returns an `Ok("Pong")` still works.

My `build.sbt` dependencies:

```scala
val Http4sVersion         = "0.20.0-M2"
val CirceVersion          = "0.10.0"
val MonixVersion          = "3.0.0-RC2"
val LogbackVersion        = "1.2.3"
val TypesafeConfigVersion = "1.3.2"
```

**Comment by rossabaker** [comment]
This is probably a bug in blaze, but we'll track it here.

Which JDK version?  Something changed with the SSLEngine in JDK 11 that is causing blaze problems, but I've not seen this.

**Comment by alexandru** [comment]
It's running on OpenJDK 8, Ubuntu (latest LTS).

**Comment by alexandru** [comment]
Btw, anything I can do to make it work? Alternatives to Blaze?

**Comment by rossabaker** [comment]
There is a servlet backend.  http4s-jetty and http4s-tomcat provide embedded containers similar to the `BlazeServerBuilder`.  They still take `HttpRoutes` instead of `HttpApp`, which is an inconsistency that's on my todo list.  If you've got an `HttpRoutes`, don't call `.orNotFound`, and you'll still have an `HttpRoutes`.

Can you expand on routes that are "actually doing something"?  Large request bodies?  Large response bodies?  Any exmaple code that could be shared and might help reproduce it?

**Comment by rossabaker** [comment]
I think I can fix this with a blaze milestone.  Watch this space.

**Comment by alexandru** [comment]
The server's code is here: https://github.com/alexandru/vimeo-download-plus/tree/bug

I created a tag to ensure you have the code. 

The server is up and running at `videos.monix.io`. It's not very deterministic. It comes and goes.

Problematic endpoints:

- https://videos.monix.io/get/299709216/
- https://videos.monix.io/redirect/299709216/SD%20360p/

When I'm testing these right now, they work. Don't abuse the links or they'll block me — in case you're wondering what this server is doing, well, it's to gain access to the raw video files without upgrading from Vimeo Plus to Pro 🙂

**Comment by rossabaker** [comment]
Try adding the following dependency.  The one it evicts is binary compatible.  I tried to simplify the code and made a bad assumption, and this reverts it to the way older blazes worked:

```
"org.http4s" %% "blaze-http" % "0.14.0-M11"
```

It's on oss.sonatype.org now, and will be on Maven Central when it syncs.

A version at least this high will be in the next http4s release, so remove this explicit dependency on the next upgrade.

**Comment by alexandru** [comment]
😲 that was fast 👍 will upgrade 🤞

**Comment by rossabaker** [comment]
I think this is resolved by 0.14.0-M11.  If not, please reopen.

**Comment by alexandru** [comment]
I haven’t noticed the problem anymore, thanks.
